hazelcast-client/src/main/java/com/hazelcast/client/config/ClientConfig.java

```diff
@@ -287,8 +288,9 @@ public ClientConfig addProxyFactoryConfig(ProxyFactoryConfig proxyFactoryConfig)
     public NearCacheConfig getNearCacheConfig(String name) {
         NearCacheConfig nearCacheConfig = lookupByPattern(nearCacheConfigMap, name);
         if (nearCacheConfig == null) {
-            nearCacheConfig = nearCacheConfigMap.get("default");
+            nearCacheConfig = new NearCacheConfig(nearCacheConfigMap.get("default"));
```

Comment:
Peter notices that it's dangerous to use the `default` configuration without cloning, since we would return the same instance for different names. This should fix that (good finding!).